processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -320,8 +321,21 @@ else if ( method.isMapMapping() ) {
                 hasFactoryMethod = mapMappingMethod.getFactoryMethod() != null;
                 mappingMethods.add( mapMappingMethod );
             }
+            else if ( method.isValueMapping() ) {
+                // prefer value mappings over enum mapping
+                ValueMappingMethod valueMappingMethod = new ValueMappingMethod.Builder()
+                    .mappingContext( mappingContext )
+                    .souceMethod( method )
+                    .valueMappings( mappingOptions.getValueMappings() )
+                    .build();
+                mappingMethods.add( valueMappingMethod );
+            }
             else if ( method.isEnumMapping() ) {
 
+                messager.printMessage(
```

Comment:
Did you checkout the UT in the old enum package? Each test case there has: 

```java
    @ExpectedCompilationOutcome(
        value = CompilationResult.SUCCEEDED,
        diagnostics = {
            @Diagnostic(type = OrderMapper.class,
                kind = Kind.WARNING,
                line = 41,
                messageRegExp = "Mapping of Enums via @Mapping is going to be removed in future versions of "
                    + "MapStruct\\. Please use @ValueMapping instead!")
        }
    )
```

So I reckon all old test cases test this implicitly.